tests/scripts/parse_notice.py

```diff
@@ -0,0 +1,277 @@
+                            f"[-] Nothing has been found for package '{real_package_name}' in {self.scanned_json_file}",
+                        )
+                        continue
+
+                    if (
+                        "package_name" not in processed_package
+                        or "version" not in processed_package
+                        or "homepage_url" not in processed_package
+                        or "license_name" not in processed_package
+                        or "license_path" not in processed_package
+                        or "license_content" not in processed_package
+                    ):
+                        print(f"Missing data for '{real_package_name}'. Skipping...")
+                        continue
+
+                    self.write_to_file(processed_package)
+                    print(f"[+] Package '{real_package_name}' has been added to {self.notice_file_name}")
+        else:
+            print("[-] Invalid argument. Please choose an argument between 'fix' or 'check'")
+            sys.exit(1)
+
+    def process_package(self, required_package: str) -> None:
+        """
+        Iterates over the json file outputted by scancode and tries to find a match between
+        the required package and the installed one and looks in the self.POSSIBLE_LICENSE_FILES
+        and self.POSSIBLE_METADATA_FILES where the important information about the package should exist
+        """
+        for entry in self.scanned_results_json["files"]:
+            splitted_entry_path: list[str] = entry["path"].split("/")
+
+            if (
+                splitted_entry_path[-1] in NoticeParser.POSSIBLE_LICENSE_FILES
+                or splitted_entry_path[-1] in NoticeParser.POSSIBLE_METADATA_FILES
+            ):
+                if not len(splitted_entry_path) > 6:
+                    continue
+
+                package_name = splitted_entry_path[5].split("-")[0]
+
+                if package_name == required_package:
+                    if package_name not in self.processed_packages:
+                        # print(entry)
+                        self.processed_packages[package_name] = {}
+                        self.processed_packages[package_name]["package_name"] = self.required_packages[required_package]
+
+                        if entry["licenses"] and len(entry["licenses"]) > 0:
+                            self.processed_packages[package_name]["license_name"] = entry["licenses"][0]["key"].upper()
+
+                    if splitted_entry_path[-1] in NoticeParser.POSSIBLE_METADATA_FILES:
+                        package_version: str = splitted_entry_path[5].split("-")[1].strip(".dist")
+                        self.processed_packages[package_name]["version"] = package_version
+
+                        homepage_url: str = entry["packages"][0]["homepage_url"]
+                        vcs_url: str = entry["packages"][0]["vcs_url"]
+
+                        if not homepage_url and not vcs_url:
```

Comment:
can we have a license but not the `Homepage URL` or `VCS URL`?
In this case we should not skip the entry but just omit the `homepage_url` value